lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2698,13 +2705,13 @@ func (d *zfs) MountVolumeSnapshot(snapVol Volume, op *operations.Operation) erro
 		if !filesystem.IsMountPoint(mountPath) {
 			err := snapVol.EnsureMountPath()
 			if err != nil {
-				return err
+				return nil, err
 			}
 
 			// Mount the snapshot directly (not possible through tools).
-			err = TryMount(snapshotDataset, mountPath, "zfs", 0, "")
+			err = TryMount(snapshotDataset, mountPath, "zfs", unix.MS_RDONLY, "")
```

Comment:
It feels like this change for unix.MS_RDONLY and presumably the ext4 bits below should be a separate commit from the splitting of the function (and the reverter being added), accompanied by a commit message explaining *why* its needed.